aioesphomeapi/connection.py

```diff
@@ -475,19 +467,14 @@ async def _connect_hello_login(self, login: bool) -> None:
             messages.append(self._make_connect_request())
             msg_types.append(ConnectResponse)
 
-        try:
-            responses = await self.send_messages_await_response_complex(
-                tuple(messages),
-                None,
-                lambda resp: type(resp)  # pylint: disable=unidiomatic-typecheck
-                is msg_types[-1],
-                tuple(msg_types),
-                CONNECT_REQUEST_TIMEOUT,
-            )
-        except TimeoutAPIError as err:
-            self.report_fatal_error(err)
-            raise TimeoutAPIError("Hello timed out") from err
```

Comment:
This was a useless try-except-raise that actually produced a worse error message